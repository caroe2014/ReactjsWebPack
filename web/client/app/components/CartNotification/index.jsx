// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import { Flex, Text } from 'rebass';
import styled from 'styled-components';
import { Trans } from 'react-i18next';
import i18n from 'web/client/i18n';
import Announcements from 'web/client/app/components/Announcements';

const TopBox = styled(Flex)`
  Flex: 0.6 1 auto;
  width: 100%;
  padding: 15px 10px;
  margin: auto;
  flex-direction: column;
  align-items: center;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const BottomBox = styled(Flex)`
  Flex: 0.4 1 auto;
  width: 100%;
  height: 2em;
  margin: auto;
  flex-direction: row;
`;

const ItemName = styled(Text)`
  color: ${props => props.theme.colors.primaryTextColor};
  width: 100%;
  font-size: 20px;
  font-weight: 500;
  margin: auto;
  text-align: center;
  word-break: break-word;
`;

const ToCartText = styled(Text)`
  font-size: 20px;
  color: ${props => props.theme.colors.primaryTextColor};
  font-weight: 400;
  margin: auto;
`;

const ButtonNotify = styled(Flex)`
  width: 100%;
  height: 100%;
  &:hover {
    color: ${props => props.theme.colors.buttonControlColor};
  }
  color: ${props => props.theme.colors.buttonControlColor};
  cursor: pointer;
  margin: auto;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const ButtonText = styled(Text)`
  font-size: ${props => props.theme.fontSize.nm};
  text-transform: uppercase;
  margin: auto;
`;

const Container = styled(Flex)`
  flex-direction: column;
  font-size: ${props => props.theme.fontSize.nm};
  height:100%;
  width:100%;
  margin: 0px !important;
  padding: 10px;

`;

class CartNotification extends Component {

  handleDelete = () => {
    if (this.props.items) {
      var lastItem = this.props.items.find(item => {
        return item.uniqueId === this.props.lastItemAdded.uniqueId;
      });
      this.props.removeItem(lastItem);
    }
    this.props.closeCartNotification(!this.props.closeNotificationFlag);
  }

  handleClose = () => {
    this.props.closeCart(!this.props.closeFlag);
    this.props.toggleCart(true);
  }

  render () {
    const { lastItemAdded, fillContent } = this.props;
    return (
      lastItemAdded &&
      <Container
        className='cart-notification'
        m={3}
        alignItems='center'
        width={fillContent ? 1 : [1, 2 / 3, 2 / 5]}
        flexDirection='column'
      >
        <Flex className='outer-container' flexDirection='column' width={1}>
          <Announcements message={`${lastItemAdded.displayText} ${i18n.t('CART_NOTIFICATION_ADDED')}`} />
          <TopBox className='top-container' tabIndex={0}
            aria-label={`${lastItemAdded.displayText} ${i18n.t('CART_NOTIFICATION_ADDED')}`}>
            <ItemName className='item-text' >
              {lastItemAdded.displayText}</ItemName>
            <ToCartText className='item-added-text'>
              <Trans i18nKey='CART_NOTIFICATION_ADDED'/>
            </ToCartText>
          </TopBox>
          <BottomBox className='bottom-container'>
            <ButtonNotify
              className='undo-button'
              onClick={this.handleDelete}
              role='button'
              tabIndex={0}
              aria-label={i18n.t('CART_NOTIFICATION_UNDO')}
            >
              <ButtonText><Trans i18nKey='CART_NOTIFICATION_UNDO'/></ButtonText>
            </ButtonNotify>
            <ButtonNotify
              className='view-cart-button'
              role='button'
              tabIndex={0}
              aria-label={i18n.t('CART_NOTIFICATION_VIEW_CART')}
              onClick={this.handleClose}>
              <ButtonText><Trans i18nKey='CART_NOTIFICATION_VIEW_CART'/></ButtonText>
            </ButtonNotify>
          </BottomBox>
        </Flex>
      </Container>
    );
  }
}

export default CartNotification;
