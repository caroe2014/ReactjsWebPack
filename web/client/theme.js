// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
