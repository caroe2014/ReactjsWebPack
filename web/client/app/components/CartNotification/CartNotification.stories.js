// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';

import { Provider } from 'rebass';

import theme from 'web/client/theme.js';
import CartNotification from './';
import config from 'app.config';
import { BrowserRouter as Router, Route } from 'react-router-dom';

const lastItemAdded = { name: 'Burger' };

storiesOf('CartNotification', module)
  .addDecorator(withKnobs)
  .add('CartNotification Component', () => {
    return (
      <div style={{ textAlign: '-webkit-center' }}>
        <Provider theme={theme} style={{ width: '30em' }}>
          <Router basename={config.webPaths.base}>
            <Route render={({ match, location, history }) => (
              <CartNotification fillContent lastItemAdded={lastItemAdded}/>
            )} />
          </Router>
        </Provider>
      </div>
    );
  });
