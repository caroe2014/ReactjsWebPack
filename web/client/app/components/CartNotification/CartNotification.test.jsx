// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import {
  shallow,
  mount,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {
  ThemeProvider
} from 'styled-components';
import theme from 'web/client/theme';
import { t } from 'i18next';

import CartNotification from '.';
configure({
  adapter: new Adapter()
});

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));

jest.mock('i18next');
t.mockImplementation((key) => {
  return key;
});

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));


describe('CartNotification', () => {

  it('does a snapshot test', () => {
    const wrapper = shallow(<CartNotification/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(shallow(<CartNotification />).exists()).toBe(true);
  });

  
  it('should render without throwing an error ob mount', () => {
    const mountWrapper = mount(
      <ThemeProvider theme={theme}>
        <CartNotification lastItemAdded/>
      </ThemeProvider>
    );
    expect(mountWrapper.exists()).toBe(true);
  });

  it('should call handleClose with spy', () => {
    let mockFn = jest.fn();
    let wrapper = shallow(<CartNotification lastItemAdded closeCart={mockFn} toggleCart={mockFn}/>);
    const spy = jest.spyOn(wrapper.instance(), 'handleClose');
    wrapper.instance().forceUpdate();
    wrapper.find('.view-cart-button').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call handleDelete with spy', () => {
    let lastItemAdded={
      uniqueId: 1001
    };
    let items=[
      {
        uniqueId: 1001
      }
    ];
    let mockFn = jest.fn();
    let wrapper = shallow(<CartNotification lastItemAdded closeCart={mockFn} items={items} lastItemAdded={lastItemAdded} closeCartNotification={mockFn} removeItem={mockFn}/>);
    const spy = jest.spyOn(wrapper.instance(), 'handleDelete');
    wrapper.instance().forceUpdate();
    wrapper.find('.undo-button').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call handleDelete with spy', () => {
    let lastItemAdded={
      uniqueId: 1001
    };
    let items=[
      {
        uniqueId: 1001
      }
    ];
    let mockFn = jest.fn();
    let wrapper = shallow(<CartNotification lastItemAdded closeCart={mockFn} fillContent lastItemAdded={lastItemAdded} closeCartNotification={mockFn} removeItem={mockFn}/>);
    const spy = jest.spyOn(wrapper.instance(), 'handleDelete');
    wrapper.instance().forceUpdate();
    wrapper.find('.undo-button').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });


});
