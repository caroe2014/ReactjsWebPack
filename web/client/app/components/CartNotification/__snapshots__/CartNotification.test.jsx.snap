// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartNotification does a snapshot test 1`] = `""`;
